api/app/Rules/SkillLevelRequiredIfEssential.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Rules;
+
+use App\Enums\PoolSkillType;
+use App\Models\JobPosterTemplate;
+use Closure;
+use Illuminate\Contracts\Validation\DataAwareRule;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+
+/**
+ * This rule is for ensuring a skills requiredLevel in a job template is
+ * required (or prohibited) based on if it's essential
+ */
+class SkillLevelRequiredIfEssential implements DataAwareRule, ValidationRule
+{
+    /**
+     * Indicates whether the rule should be implicit.
+     *
+     * @var bool
+     */
+    public $implicit = true;
+
+    /**
+     * All of the data under validation.
+     *
+     * @var array<string, mixed>
+     */
+    protected $data = [];
+
+    /**
+     * Create a new rule instance.
+     *
+     * @return void
+     */
+    public function __construct() {}
+
+    /**
+     * Run the validation rule.
+     *
+     * @param  \Closure(string): \Illuminate\Translation\PotentiallyTranslatedString  $fail
+     */
+    public function validate(string $attribute, mixed $incomingLevel, Closure $fail): void
+    {
+        $incomingType = Arr::get($this->data, Str::beforeLast($attribute, '.').'.type');
+        Log::debug($this->data);
```

Comment:
You probably don't need this anymore?

Reply:
Removed in [c288609](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11543/commits/c288609bbac5ce2a5ea65326889b9b8982c004aa).